tests/rptest/tests/consumer_group_test.py

```diff
@@ -639,3 +646,149 @@ class OffsetFetchRequest_v5(Request):
+                if msg.error():
+                    self.logger.error(
+                        f"consumer {self.id} error - {msg.error()}")
+                    continue
+
+                self.consumer.store_offsets(msg)
+                with self.lock:
+                    self.last_consumed = msg.offset()
+
+            except Exception as e:
+                self.logger.error(f"consumer {self.id} error - {e}")
+
+        self.consumer.close()
+
+    def get_last_consumed(self):
+        with self.lock:
+            return self.last_consumed
+
+    def restart_consumer(self):
+        self.restart.set()
+        self.restarted.wait()
+        self.restarted.clear()
+
+
+class ConsumerGroupStaticMembersRebalance(RedpandaTest):
+    @cluster(num_nodes=4)
+    def test_rejoining(self):
+        topic = TopicSpec(name="test-topic-1", partition_count=20)
+        DefaultClient(self.redpanda).create_topic(topic)
+        group_id = "test-group-1"
+
+        producer = KgoVerifierProducer(self.test_context,
+                                       self.redpanda,
+                                       topic.name,
+                                       msg_size=128,
+                                       msg_count=5000000)
+        producer.start()
+        consumers = []
+
+        for c_id in range(20):
+            consumers.append(
+                TestConsumer(bootstrap_servers=self.redpanda.brokers(),
+                             group=group_id,
+                             topic=topic.name,
+                             id=c_id,
+                             logger=self.logger))
+
+        admin_client = AdminClient(
+            {"bootstrap.servers": self.redpanda.brokers()})
+
+        def get_group_description():
+            description = admin_client.describe_consumer_groups(
+                group_ids=[group_id])[group_id].result()
+            return description
+
+        def consumers_made_progress():
+            return all(c.get_last_consumed() is not None for c in consumers)
+
+        wait_until(consumers_made_progress, 60, 1)
+        before_restart = get_group_description()
+        self.logger.info("group state before restart: %s",
+                         before_restart.state)
+        assert before_restart.state == ConsumerGroupState.STABLE
+        assert len(before_restart.members) == 20
+        consumer_to_restart = random.choice(consumers)
+        last_consumed = consumer_to_restart.get_last_consumed()
+        consumer_to_restart.restart_consumer()
+        after_restart = get_group_description()
+        self.logger.info("group state after restart: %s", after_restart.state)
+        assert after_restart.state == ConsumerGroupState.STABLE
+        assert len(after_restart.members) == 20
+
+        wait_until(
+            lambda: consumer_to_restart.get_last_consumed() > last_consumed,
+            60, 1)
+        # now stop consumer, after a timeout group should rebalance and the member should be removed
```

Comment:
Is this timeout hardcoded/configurable? Could you point me to it, please? I'm asking partly to learn and partly to see if it's worth adjusting that config for the sake of this test.